src/roman_datamodels/maker_utils/_datamodels.py

```diff
@@ -430,3 +431,53 @@ def mk_guidewindow(*, shape=(2, 8, 16, 32, 32), filepath=None, **kwargs):
     guidewindow["amp33"] = kwargs.get("amp33", u.Quantity(np.zeros(shape, dtype=np.uint16), u.DN, dtype=np.uint16))
 
     return save_node(guidewindow, filepath=filepath)
+
+
+def mk_source_catalog(*, filepath=None, **kwargs):
+    """
+    Create a dummy Source Catalog instance (or file) with arrays and valid values
+    for attributes required by the schema.
+
+    Parameters
+    ----------
+    filepath
+        (optional, keyword-only) File name and path to write model to.
+
+    Returns
+    -------
+    roman_datamodels.stnode.SourceCatalog
+    """
+    source_catalog = stnode.SourceCatalog()
+
+    source_catalog["source_catalog"] = kwargs.get("source_catalog", Table([range(3), range(3)], names=["a", "b"]))
+    source_catalog["meta"] = mk_common_meta()
+    source_catalog["meta"].update(kwargs.get("meta", dict(segmentation_map="")))
+
+    return save_node(source_catalog, filepath=filepath)
+
+
+def mk_segmentation_map(*, filepath=None, shape=(4096, 4096), **kwargs):
```

Comment:
Is the segmentation map going to be a data model in its own right?

Reply:
Yes.